### DISCUSSION TITLE
Lab 06.02 Task3

### DISCUSSION THREAD

**Discussion by chape8** [rramosp/ai4eng.v1#208] [06 - MÉTODOS NO SUPERVISADOS]
Estoy realizando la ultima tarea donde al mandarla al calificador me aparece erróneo el resultado donde siento equivocación por parte del calificador.Este es el error que sale 
![erorr lb6](https://user-images.githubusercontent.com/98787315/199561932-802787b9-f1ca-4c33-9d41-0c0502b5639f.png)

Con los vectores X y v que dicen que están probando hago lo mismo y según mi análisis el código está funcionando bien por que según los resultado de distancias: [23.91652149, 0.0, 24.4744765, 23.36664289, 23.10844002],  al aplicar np.argsort debería ser el que precisamente muestra: [1 4 3 0 2], cosa que contradice lo que según espera que salga como resultado en el calificador

![6 02 x_v](https://user-images.githubusercontent.com/98787315/199562490-90abf14d-d3b5-4ca5-9665-41501a5bab08.png)

**Comment by imEag** [comment]
Hola @chape8 Lo primero que te recomiendo es que en el espacio [49] líneas 7 y 8 no utilices el método `fit_transform` sino que utilices el método `transform` porque el primero hace que el modelo aprenda de X y luego transforme los datos pero en la línea 6 ya habías aplicado el método `fit` que tiene la función de hacer que el modelo aprenda de los datos.

Tampoco es necesario que utilices `np.round` (línea 13).

Por último no olvides utilizar la forma que te recomiendan en el notebook


```
def find_similar(v,X,n):
    from sklearn.decomposition import NMF
    
    nmf = NMF(n_components=30, init="random", random_state=0)
    nmf... ## your code here. call the 'fit' method
    Xt = ... # use nmf to transform X
    vt = ... # use nmf to transform v .. you will have to use reshape like this v.reshape(1,-1)
    
    result = ... # your code here
    
    return result
```


Inténtalo y me cuentas como te va.

**Reply by chape8** [reply]
Hola [imEag](https://github.com/imEag), ya me dio haciendo lo que me recomendaste. Además, caí en cuenta de que estaba utilizando X y v sin transformar entonces reemplacé en la linea 11 X por Xt y el la 12 v por vt. Muchas gracias

**Reply by imEag** [reply]
Perfecto!
